### ISSUE TITLE
CA md too weak when following mosquitto TLS guide

### ISSUE THREAD

**Issue by mike-sandstrom** [eclipse/mosquitto#2135]
I have followed the mosquitto tls steps to self signing tls certificates and ran in to the problem of my server certificate being too weak. I dont know if this error is thrown only on newer OS distributions (or newer openssl versions)- I am using Ubuntu 20.1 and OpenSSL 1.1.1f  31 Mar 2020. 

The Error:
`1615839836: Error: Unable to load server certificate "/etc/mosquitto/certs/server.crt". Check certfile.`
`1615839836: OpenSSL Error[0]: error:140AB18E:SSL routines:SSL_CTX_use_certificate:ca md too weak`

The Fix:
I fixed this error by regenerating the server.crt with sha256-- I checked with openssl and server.crt was originally created with sha1 by default, which is the source of the error. It would probably be a good idea to make note of this in the mosquitto-tls manual. 

openssl x509 -req -in server.csr -CA ca.crt -CAkey ca.key -CAcreateserial -out server2.crt **-sha256** -days 360

Question:
If this now creates a problem of too much overhead when using TLS--could I run the same command but using some elliptic curve algorithm which I have heard is a more lightweight? Would this work if the ca_certificate is created with sha256? If not--I can redo the tls process and create keys with ECC algorithms to reduce some overhead. Thank you!

**Comment by mike-sandstrom** [comment]
While the above was able to get mosquitto to start, I receive this error when trying to connect with paho-mqtt python library. 
`ConnectionResetError: [Errno 104] Connection reset by peer`

I used the ip as the common name for the server certificate. If I use the domain name it wont work because my domain is behind cloudflare, which doesnt allow mqtt ports to pass through their network. I dont know if this is related at all.

**Comment by ralight** [comment]
To be honest, my first reaction to this is "wow, you found an openssl install that still generated sha1 signed certificates by default". The oldest system I have easy access to is Ubuntu 18.04 and that uses sha256 by default. Browsers stopped allowing sha1 years ago.

I would suggest starting on a single machine using the certificates in https://github.com/eclipse/mosquitto/tree/master/test/ssl for your testing - give all-ca.crt to the python client as the CA certificate and connect to localhost as the hostname. Use server.crt and server.key for the broker. If that works, you know that you have everything else working and it is "just" a case of getting the certificates correct - which can be a big "just" unfortunately.

**Comment by mike-sandstrom** [comment]
yes, this can be closed apologies for not following up with this. I mistakenly stated I was using openssl when I actually had LibreSSL 2.8.3 installed on the machine used to generate the certificates. I created the certificates after with the correct openssl version and the handshake was successful.
